ehsm_kms_service/secret_manager_apis.js

```diff
@@ -151,6 +152,60 @@ function calculateRotationSperate(rotationInterval, createTime) {
     return nextRotationDate
 }
 
+/**
+ * Force delete secret metadata and secret version data by appId and secretName
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @returns {boolean}
+ */
+const forceDeleteData = async (DB, appid, secretName) => {
+    try {
+        //Query and delete the secret metadata
+        const query_secret_metadata = {
+            selector: {
+                appid,
+                secretName
+            },
+            fields: ['_id', '_rev'],
+            limit: 1
+        }
+        let secret_metadata_res = await DB.partitionedFind('secret_metadata', query_secret_metadata)
+        if (secret_metadata_res.docs.length > 0) {
+            await DB.destroy(secret_metadata_res.docs[0]._id, secret_metadata_res.docs[0]._rev)
+            //Query and delete the secret version data (1000 queries per time)
+            let needDelete = true
+            while (needDelete) {
+                const query_secret_version_data = {
+                    selector: {
+                        appid,
+                        secretName
+                    },
+                    fields: ['_id', '_rev'],
+                    limit: 1000
+                }
+                let secret_version_data_res = await DB.partitionedFind('secret_version_data', query_secret_version_data)
+                if (secret_version_data_res.docs.length > 0) {
+                    for (const secret_version_data_item of secret_version_data_res.docs) {
+                        secret_version_data_item._deleted = true
```

Comment:
forceDeleteData only delete metadata but not version data?

Reply:
it will delete metadata and all version data by secretname and appid